### ISSUE TITLE
Calculate test loss as a function of training epoch

### ISSUE THREAD

**Issue by htjb** [htjb/globalemu#10]
This should be a useful and easy addition to the code to help determine whether the code is training properly. Runtime comparison of the test and training losses can help determine overfitting and can be used to implement a proper early stopping function.
